hudi-common/src/main/java/org/apache/hudi/common/util/HoodieTimer.java

```diff
@@ -30,7 +30,17 @@
 public class HoodieTimer {
 
   // Ordered stack of TimeInfo's to make sure stopping the timer returns the correct elapsed time
-  Deque<TimeInfo> timeInfoDeque = new ArrayDeque<>();
+  private final Deque<TimeInfo> timeInfoDeque = new ArrayDeque<>();
+
+  public HoodieTimer() {
+    this(false);
+  }
+
+  public HoodieTimer(boolean shouldStart) {
+    if (shouldStart) {
+      startTimer();
+    }
+  }
```

Comment:
Old semantic is still preserved: it works as it have been, and just adds new way when you don't need to invoke `startTimer` explicitly

Reply:
Understood.  I'm saying that `timer = new HoodieTimer().startTimer()` looks obvious for starting the timer instead of looking into what the boolean represents.

Reply:
Fair enough